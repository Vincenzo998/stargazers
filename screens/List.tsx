import React, { useState } from "react";
import { VirtualizedList } from "react-native";

// hooks
import { useStargazers } from "../hooks";

// components
import SkeletonList from "../components/SkeletonList";
import ListItem from "../components/ListItem";

// types
import { Item, Info, ListItemProps } from "../types";

// defines
import { REPO, USER } from "../defines";

// styles
import { Title } from "../styles/common";
import { ListContainer, InputContainer, Input, Label, ErrorText } from "./List.style";

const List: React.FC = () => {
  const [info, setInfo] = useState<Info>({
    [USER]: "",
    [REPO]: "",
  });

  const { stargazers, loading, error } = useStargazers(info);

  const handleInfo = (key: string, value: string) => setInfo((prevInfo) => ({ ...prevInfo, [key]: value }));

  return (
    <ListContainer>
      <Title> Stargazers </Title>
      <InputContainer style={{ marginBottom: 15 }}>
        <Label> User </Label>
        <Input onChangeText={(value: string) => handleInfo(USER, value)} value={info[USER]} placeholder="Enter a user" />
      </InputContainer>
      <InputContainer>
        <Label> Repository </Label>
        <Input onChangeText={(value: string) => handleInfo(REPO, value)} value={info[REPO]} placeholder="Enter a repository" />
      </InputContainer>
      {loading ? (
        <SkeletonList />
      ) : error ? (
        <ErrorText> {error} </ErrorText>
      ) : (
        <VirtualizedList
          data={stargazers}
          initialNumToRender={6}
          keyExtractor={(_: {}, index: number) => index.toString()}
          renderItem={({ item, index }: ListItemProps) => <ListItem index={index} item={item} />}
          getItem={(data: Array<Item>, index: number): Item => data[index]}
          getItemCount={(data: []) => data.length}
        />
      )}
    </ListContainer>
  );
};

export default List;
