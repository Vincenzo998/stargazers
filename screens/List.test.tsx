import React from "react";
import renderer from "react-test-renderer";

// screens
import List from ".";

test("List should render without crashing", () => {
  const ListComponent = renderer.create(<List />).toJSON();
  expect(ListComponent).toMatchSnapshot();
});
