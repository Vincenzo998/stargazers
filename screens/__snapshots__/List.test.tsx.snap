// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List should render without crashing 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#fff",
        "flexBasis": 0,
        "flexDirection": "column",
        "flexGrow": 1,
        "flexShrink": 1,
        "paddingVertical": "15%",
      },
    ]
  }
>
  <Text
    style={
      [
        {
          "fontFamily": "Al Nile",
          "fontSize": 20,
          "fontWeight": "bold",
          "textAlign": "center",
        },
      ]
    }
  >
     Stargazers 
  </Text>
  <View
    style={
      [
        {
          "justifyContent": "space-evenly",
          "marginHorizontal": "5%",
        },
        {
          "marginBottom": 15,
        },
      ]
    }
  >
    <Text
      style={
        [
          {
            "fontFamily": "Al Nile",
          },
        ]
      }
    >
       User 
    </Text>
    <TextInput
      onChangeText={[Function]}
      placeholder="Enter a user"
      style={
        [
          {
            "borderColor": "#656565",
            "borderRadius": "13%",
            "borderStyle": "solid",
            "borderWidth": 1,
            "paddingBottom": "5%",
            "paddingLeft": "5%",
            "paddingRight": "5%",
            "paddingTop": "5%",
            "width": "100%",
          },
        ]
      }
      value=""
    />
  </View>
  <View
    style={
      [
        {
          "justifyContent": "space-evenly",
          "marginHorizontal": "5%",
        },
      ]
    }
  >
    <Text
      style={
        [
          {
            "fontFamily": "Al Nile",
          },
        ]
      }
    >
       Repository 
    </Text>
    <TextInput
      onChangeText={[Function]}
      placeholder="Enter a repository"
      style={
        [
          {
            "borderColor": "#656565",
            "borderRadius": "13%",
            "borderStyle": "solid",
            "borderWidth": 1,
            "paddingBottom": "5%",
            "paddingLeft": "5%",
            "paddingRight": "5%",
            "paddingTop": "5%",
            "width": "100%",
          },
        ]
      }
      value=""
    />
  </View>
  <RCTScrollView
    data={[]}
    getItem={[Function]}
    getItemCount={[Function]}
    initialNumToRender={6}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={[]}
  >
    <View />
  </RCTScrollView>
</View>
`;
